common/src/main/java/io/netty/util/internal/ObjectUtil.java

```diff
@@ -73,12 +138,34 @@ public static int checkPositiveOrZero(int i, String name) {
      * Otherwise, returns the argument.
      */
     public static long checkPositiveOrZero(long l, String name) {
-        if (l < 0) {
-            throw new IllegalArgumentException(name + ": " + l + " (expected: >= 0)");
+        if (l < LONG_ZERO) {
+            throw new IllegalArgumentException(String.format(EX_GREATER_EQUAL_ZERO_DECIMAL, String.valueOf(name), l));
         }
         return l;
     }
 
+    /**
+     * Checks that the given argument is positive or zero. If it is not, throws {@link IllegalArgumentException}.
+     * Otherwise, returns the argument.
+     */
+    public static double checkPositiveOrZero(final double d, final String name) {
+        if (d < DOUBLE_ZERO) {
+            throw new IllegalArgumentException(String.format(EX_GREATER_EQUAL_ZERO_FLOAT, String.valueOf(name), d));
```

Comment:
Is it ok ? That we use `%f`  for double without any precision, it can leads different value in exception and original double. e.g origin val: 1.12344675 after format: 1.123447.

Reply:
@amizurov I was wondering about that one, too. But [java.util.Formatter](https://docs.oracle.com/javase/8/docs/api/index.html?java/util/Formatter.html) has not a special option here. A quote from the javadoc 

> Floating Point - may be applied to Java floating-point types: float, Float, double, Double, and BigDecimal 

I will not not use String.format for ease of test reason at all, see comment above.

Reply:
Ok, thanks